### PR TITLE
Add unit test using jest and add configuration to create coverage report

### PR DESCRIPTION
**Pull request by monikaprabhu**
-  Add unit test using jest and enzyme to simulate  mouse events and check the status of the event on libraryUI
    Create a component "LibraryItem"
    Simulate "click" using enzyme which is utility to testing assert on the component.
    Verify the state of the component changes to "enabled" following mouse click.
-  Configuration to create coverage report on jest tests
    Add command to "package.json" for creation of coverage reports for jest tests.
    To create coverage reports run command : "npm run testcover".
-   Wiki for setting up test framework and adding tests is found here - 
     https://github.com/DynamoDS/librarie.js/wiki/Setting-up-test-frameworks-for-librarie.js
-  Add example test to  Readme.md for test

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### CODE REVIEW

**Comment by monikaprabhu** [comment]
@Benglin @fanwgwg - please review and merge

File: README.md [hunk 1]
```diff
@@ -219,6 +222,24 @@ libController.on(libController.ItemMouseLeaveEventName, function(contextData) {
...
+          let libContainer = libController.createLibraryContainer(layoutSpecsJson, loadedTypesJson);
+
+          let libraryItem = mount(<LibraryItem libraryContainer={libContainer} data={data} />); 
+          expect(libraryItem.props().data.childItems[0].text).to.equal("Child0"); 
+
+}
+```
+- Instructions to write a tests are found at https://github.com/DynamoDS/librarie.js/wiki/Setting-up-test-frameworks-for-librarie.js
+
+
```
Comment (Benglin) [inline README.md]: Why is this section placed in between two `Event` sections?

File: README.md [hunk 2]
```diff
@@ -19,7 +19,10 @@ Installing all dependencies
 
 - Navigate to `localhost:3456` in Google Chrome browser
 
-## Test
+## Setup Testframework
+- Step by step instructions to setup JEST is found at https://github.com/DynamoDS/librarie.js/wiki/Setting-up-test-frameworks-for-librarie.js
+
+## Run Tests
```
Comment (Benglin) [inline README.md]: Previous comments in [another PR](https://github.com/DynamoDS/librarie.js/pull/62) are not addressed.

File: __tests__/jesttest.tsx [hunk 3]
```diff
@@ -92,6 +94,43 @@ describe("LibraryContainer", function () {
     expect(libraryItem.prop('data').childItems[0].childItems[0].text).to.equal("Child1");
     expect(libraryItem.prop('data').childItems[0].childItems[1].text).to.equal("Child2");
   });
+   describe("Test mouse click event and change in expand state", function () {
```
Comment (ellensi) [inline __tests__/jesttest.tsx]: @monikaprabhu Please help to extract this `describe()` block to be on the same level with the other `describe()` blocks in the same file, as we have clarified that we have no intention to nest the blocks.

I will add my new `describe()` and `it()` blocks accordingly after your change.

Thanks :wink:

File: README.md (hunk 1 above)
Comment (monikaprabhu) [inline README.md]: @Benglin , I think Readme.md was modified after I created the initial PR and automatic merge moved it in between event section. I've removed it.

**Comment by monikaprabhu** [comment]
@Benglin - I've resolved the comments in the PR. Please review and merge

File: README.md (hunk 1 above)
Comment (Benglin) [inline README.md]: Thanks @monikaprabhu, understood. However, this [still does not look okay](https://github.com/monikaprabhu/librarie.js/blob/338f5d031b417ca51be047d1d180f2efab1822c3/README.md) 😢 

Can you help ensuring:

1. Documentation formatting is proper
2. Move it under `Run Tests` section above

Thanks

Comment (monikaprabhu) [inline README.md]: I've done Shift+option+F before committing this file, but does not show much change in formatting. I've moved the test under Run tests now.

**Comment by monikaprabhu** [comment]
@Benglin - all comments above are resolved - this can be merged now.

**Comment by Benglin** [comment]
Thanks so much @monikaprabhu, LGTM!

**Pull request merged**